corehq/apps/case_importer/views.py

```diff
@@ -213,6 +213,12 @@ def _process_file_and_get_upload(uploaded_file_handle, request, domain, max_colu
             'you must first create an application with a case list.'
         ))
 
+    # Remove deprecated case types as options. We only do that here as we don't want to remove
+    # deprecated case types when determing if the import is a bulk case import
+    data_dict_deprecated_case_types = get_data_dict_deprecated_case_types(domain)
+    case_types_from_apps = set(case_types_from_apps) - data_dict_deprecated_case_types
+    unrecognized_case_types = set(unrecognized_case_types) - data_dict_deprecated_case_types
```

Comment:
The `get_case_types_for_domain_es()` function does not include `CaseType` model instances. Instead, it simply fetches all the case types for which there exists at least one case. To include case types from the Data Dictionary as well,  `get_case_types_for_domain()` can be used.

After looking through the function, it would be a substantial effort to write tests for this function, given that a lot of it deals with file handling. This is therefore something that is outside the scope of this PR. Currently, the entire case import views file has no existing tests and I think it would be good to have a separate PR to create test coverage for this file.

If you're curious to see how the helper functions work, `get_case_types_for_domain_es()`  does have a unit test that you can look at. I have also added an additional test for `get_case_types_from_apps()`.